docs/development/style-guide.rst

```diff
@@ -208,49 +229,101 @@ Example
 -------
 * Data from the Gaia mission was released on 2018-04-25.
 
-A Note About Style and Tone
+A Note About Voice and Tone
 ===========================
 
 Across all Astropy materials in narrative sections, please write in:
 
-* The present tense. For example:
-  * In the following section, we are going to make a plot…
-  * To test if your version of ``astropy`` is running correctly…
+The present tense. For example:
+* In the following section, we are going to make a plot...
+* To test if your version of ``astropy`` is running correctly...
 
-* The first-person inclusive plural. For example:
-  * We did this the long way, but next let’s try the short way…
+The first-person inclusive plural. For example:
+* We did this the long way, but next we can try it the short way...
 
-* Use the generic pronoun “you” instead of “one.” For example:
-  * You can access any of the attributes on a frame by...
+Use the generic pronoun “you” instead of “one.” For example:
+* You can access any of the attributes on a frame by...
 
 Always avoid extraneous or belittling words such as “obviously,” “easily,”
 “simply,” “just,” or “straightforward.” Avoid extraneous phrases like, “we just
 have to do one more thing.”
 
+Avoid words or phrasing that create worry in the mind of the reader, such as an
+overuse of "Warnings." Instead, use positive language that establishes
```

Comment:
You are understanding it correctly! Kelle and I had seen a note like this in Python's style guide a while back and we discussed adding something like that to our style guide to discourage an overuse of warnings. Here is the note in Python's style guide: https://devguide.python.org/documenting/#affirmative-tone

What do others think of warnings vs. affirmative tone? Maybe there is a way to reword this so it's clear that we're not saying "no warnings at all" or maybe we can provide examples of when a warning is necessary and when it might create unnecessary worry in the mind of the reader?

Reply:
Thanks! The devguide link definitely clarifies it for me. I'll let others chime in as well.

Reply:
What do you think @kelle?

Reply:
Looking at our warnings, I see most are of the type where it is really about a feature not yet being complete or not always guaranteed to work as expected, which is rather different from what the python guide example is about (where indeed the reader might feel discouraged by never having thought about corruption, etc.).

I think those types of warnings are important to highlight (as @pllim noted) as they help give a sense of the limitations of the code. Indeed, they may in fact be good from the perspective of showing that, no, the developers are not perfect and try to warn the reader of the limitations of the code (and therefore their own limitation). Andnd that, of course, help to overcome these limitations is very welcome!  (More contributors!!)

So concretely, maybe another way to phrase the recommendation about the language of the warning is be sure to make clear that a warning is about a limitation in the code, not a (implied) limitation in of the reader.

Reply:
(sorry for the rambling comment - it needs some style editing!! but the point is hopefully still clear)

Reply:
> So concretely, maybe another way to phrase the recommendation about the language of the warning is be sure to make clear that a warning is about a limitation in the code, not a (implied) limitation in of the reader.

I like that clarification a lot! I'll try rewording it to make this clear.